internal/jet/array_expression.go

```diff
@@ -0,0 +1,93 @@
+package jet
+
+// ArrayExpression interface
+type ArrayExpression[E Expression] interface {
+	Expression
+
+	EQ(rhs ArrayExpression[E]) BoolExpression
+	NOT_EQ(rhs ArrayExpression[E]) BoolExpression
+	LT(rhs ArrayExpression[E]) BoolExpression
+	GT(rhs ArrayExpression[E]) BoolExpression
+	LT_EQ(rhs ArrayExpression[E]) BoolExpression
+	GT_EQ(rhs ArrayExpression[E]) BoolExpression
+
+	CONTAINS(rhs ArrayExpression[E]) BoolExpression
+	IS_CONTAINED_BY(rhs ArrayExpression[E]) BoolExpression
+	OVERLAP(rhs ArrayExpression[E]) BoolExpression
+	CONCAT(rhs ArrayExpression[E]) ArrayExpression[E]
+	CONCAT_ELEMENT(E) ArrayExpression[E]
+
+	AT(expression IntegerExpression) Expression
```

Comment:
`AT` should return element type `E`. Take a look at `rangeTypeCaster` how to extract element type.

Reply:
Done